instantlaunches/mgmt.go

```diff
@@ -0,0 +1,180 @@
+package instantlaunches
+
+import (
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+)
+
+const addInstantLaunchQuery = `
+	INSERT INTO instant_launches (quick_launch_id, added_by)
+	VALUES ( $1, ( SELECT u.id FROM users u WHERE u.username = $2 ) )
+	RETURNING id, quick_launch_id, added_by, added_on;
+`
+
+// AddInstantLaunch registers a new instant launch in the database.
+func (a *App) AddInstantLaunch(quickLaunchID, username string) (*InstantLaunch, error) {
+	newvalues := &InstantLaunch{}
+	err := a.DB.QueryRowx(addInstantLaunchQuery, quickLaunchID, username).Scan(&newvalues.ID,
+		&newvalues.QuickLaunchID,
+		&newvalues.AddedBy,
+		&newvalues.AddedOn,
+	)
+	return newvalues, err
+}
+
+// AddInstantLaunchHandler is the HTTP handler for adding a new instant launch.
+func (a *App) AddInstantLaunchHandler(c echo.Context) error {
+	il, err := NewInstantLaunchFromJSON(c.Request().Body)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "cannot parse JSON")
+	}
+
+	if il.AddedBy == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, "username was not set")
+	}
+
+	if !strings.HasSuffix(il.AddedBy, a.UserSuffix) {
+		il.AddedBy = fmt.Sprintf("%s%s", il.AddedBy, a.UserSuffix)
+	}
+
+	newil, err := a.AddInstantLaunch(il.QuickLaunchID, il.AddedBy)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return echo.NewHTTPError(http.StatusNotFound, err.Error())
+		}
+		return err
+	}
+
+	return c.JSON(http.StatusOK, newil)
+}
```

Comment:
This is nice and tidy. I like it! 👍 I occasionally find myself wishing that Go had a conditional operator, though. 😆